shared/util/saga.js

```diff
@@ -159,15 +159,15 @@ function safeTakeEveryPure<A, R, FinalAction, FinalActionError>(
 function safeTakeLatestPure<A, R, FinalAction, FinalActionError>(
   pattern: string | Array<any> | Function,
   pureWorker: ((action: A, state: TypedState) => any) | ((action: A) => any),
-  actionCreatorsWithResult?: (result: R) => FinalAction,
-  actionCreatorsWithError?: (result: R) => FinalActionError
+  actionCreatorsWithResult?: (result: R, action: A) => FinalAction,
+  actionCreatorsWithError?: (result: R, action: A) => FinalActionError
 ) {
-  return safeTakeLatest(pattern, function* safeTakeLatestPureWorker(action: A) {
+  const safeTakeLatestPureWorker = function* safeTakeLatestPureWorker(action: A) {
     // If the pureWorker fn takes two arguments, let's pass the state
     try {
       let result
       if (pureWorker.length === 2) {
-        const state: TypedState = yield select()
+        const state: TypedState = yield select(s => s)
```

Comment:
flow got very confused with the select() syntax here, so passing s=>s fixes that